misk/src/main/kotlin/misk/web/HttpActionScopeSeedDataInterceptor.kt

```diff
@@ -0,0 +1,22 @@
+package misk.web
+
+import com.google.inject.Key
+import misk.Action
+import misk.scope.ActionScope
+
+/** Allows users to seed the [ActionScope] entered when handling an HTTP request. */
+interface HttpActionScopeSeedDataInterceptor {
+  /**
+   * Accepts the current seed data, along with the [PathPattern] and [Action] for the [BoundAction]
+   * that is being handled.
+   *
+   * Returns a new seed data map which is used to enter the [ActionScope].
+   */
+  fun intercept(
+    seedData: Map<Key<*>, Any?>,
+    pathPattern: PathPattern,
+    action: Action,
+  ): Map<Key<*>, Any?>
+
+  // TODO(jeff) - Do we want this to follow the other Interceptor patterns and use a Factory?
```

Comment:
Open to going the Factory route, wasn't sure if we needed it, or if we wanted to be consistent regardless.

Reply:
If your expected implementations need to parse or dispatch on PathPattern, then it’s worthwhile going with a factory. That way you can compute and validate eagerly. Like if an annotation is malformed or whatever.